x-pack/test/functional/apps/machine_learning/anomaly_detection/saved_search_job.ts

```diff
@@ -274,12 +274,19 @@ export default function({ getService }: FtrProviderContext) {
   describe('saved search', function() {
     this.tags(['smoke', 'mlqa']);
     before(async () => {
-      await esArchiver.load('ml/farequote');
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await ml.testResources.createIndexPatternIfNeeded('ft_farequote', '@timestamp');
+      await ml.testResources.createSavedSearchFarequoteFilterIfNeeded();
+      await ml.testResources.createSavedSearchFarequoteLuceneIfNeeded();
+      await ml.testResources.createSavedSearchFarequoteKueryIfNeeded();
+      await ml.testResources.createSavedSearchFarequoteFilterAndLuceneIfNeeded();
+      await ml.testResources.createSavedSearchFarequoteFilterAndKueryIfNeeded();
+      await ml.testResources.setKibanaTimeZoneToUTC();
```

Comment:
Looks like each of these arrangements belongs to the dedicated test suite. Maybe it worth moving to `before` hook for each test suite `describe` section?

Reply:
As discussed on slack: each tests only loads the data it needs (and only if it's not already there). Moving the test resources loading to the top suite would mean that a single test execution (e.g. for debugging) loads all suite data and thus taking longer than needed.
I agree that the current test structure is a but chaotic and I plan to refactor the test suite / test / test step structure in the future, maybe things will become a bit clearer then.